### PR TITLE
Fighting flakiness: no longer uses `beCloseToDate` in `CustomerInfoOfflineEntitlementsStoreKitTest.verifyEntitlement`

### PR DESCRIPTION
**Pull request by JayShortway**
I have seen that `beCloseToDate()` / `beCloseToNow()` is one of our sources of flakiness. For instance, see the last test [here](https://app.circleci.com/pipelines/github/RevenueCat/purchases-ios/22711/workflows/611f9df0-7cef-4abe-a622-c64cb5e97f97/jobs/263013/tests). A hiccup in the CI server, an extra background process, a little bit more heat in the CircleCI data center, and things can take a fraction longer, causing this to fail.

This PR makes the tests deterministic, by defining the exact date we expect. It only tackles `CustomerInfoOfflineEntitlementsStoreKitTest.verifyEntitlement()` for now. More to come in later PRs.

### CODE REVIEW

File: Sources/OfflineEntitlements/PurchasedSK2Product.swift [hunk 1]
```diff
@@ -47,7 +47,7 @@ extension PurchasedSK2Product {
         self.entitlement = .init(
             expiresDate: expiration,
             productIdentifier: transaction.productID,
-            purchaseDate: Date(),
+            purchaseDate: transaction.purchaseDate,
```
Comment (JayShortway) [inline Sources/OfflineEntitlements/PurchasedSK2Product.swift:50]: Note that this is an actual implementation change. However, using the transaction date might be more correct? This was introduced in https://github.com/RevenueCat/purchases-ios/pull/2358, but there are no details on this particular property. Let me know if this is actually not what we want.

**Comment by JayShortway** [comment]
@RCGitBot please test

**Comment by JayShortway** [comment]
@RCGitBot please test

File: Sources/OfflineEntitlements/PurchasedSK2Product.swift (hunk 1 above)
Comment (tonidero) [reply Sources/OfflineEntitlements/PurchasedSK2Product.swift:50]: Seems this is only used for offline entitlements and conceptually it makes sense. Also checked on Android and seems this is the logic we use there so I think it's ok...

But I would like someone from @RevenueCat/catforms to double-check.

**Review by tonidero — APPROVED** [review]
This is great! Thanks for starting on this 🙏. Seems ok to me. I would like someone else from @RevenueCat/catforms to also confirm it's ok

File: Sources/OfflineEntitlements/PurchasedSK2Product.swift (hunk 1 above)
Comment (JayShortway) [reply Sources/OfflineEntitlements/PurchasedSK2Product.swift:50]: Thanks Toni! Hi @RevenueCat/catforms, could you have a quick look? 🙏

Comment (aboedo) [reply Sources/OfflineEntitlements/PurchasedSK2Product.swift:50]: probably a remnant from the first drafts where we were just trying to make it work with whatever data was available, but I don't think there's any reason (or any wins) to make this `Date()`, transaction purchase date is more accurate

**Review by aboedo — APPROVED** [review]

**Review by MarkVillacampa — APPROVED** [review]
good catch!

File: Sources/OfflineEntitlements/PurchasedSK2Product.swift (hunk 1 above)
Comment (JayShortway) [reply Sources/OfflineEntitlements/PurchasedSK2Product.swift:50]: That's good to know. Thanks for verifying!

**Pull request merged**